### DISCUSSION TITLE
What solution (if any) is the Phlex community using to sort Tailwind classes?

### DISCUSSION THREAD

**Discussion by bluz71** [yippee-fun/phlex#812] [General]
Hello folks,

I dabbled a little bit Astro with Tailwind recently and I setup my Neovim to invoke Prettier on save (with Prettier-for-Tailwind), this nicely re-ordered the Tailwind classes to the recommended order. 

I would find it difficult to live without this in Phlex.

Phlex being Ruby (aka doesn't look like tagged HTML), what are folks doing to achieve the same sorting niceness?

Just having a look around, it may be possible in Neovim land to extend the [tailwind-sorter.nvim](https://github.com/laytan/tailwind-sorter.nvim) plugin for Ruby files with Treesitter to look for `class:` nodes and do the sorting from there (maybe)? To-be-determined.

I am curious what solutions folks are using now, even for other editors, maybe even using interesting tools such as `rustywind`?

Cheers.

**Comment by bluz71** [comment]
Interesting, [rustywind](https://github.com/avencera/rustywind) (a Rust implemented Tailwind command line sorter) supports `--custom-regex` as seen in [this issue](https://github.com/avencera/rustywind/issues/96).

Folks here got the official Tailwind LSP to work with Phlex by tweaking the `regex` for the LSP as noted in [this discussion](https://github.com/orgs/phlex-ruby/discussions/536).

So maybe something similar can be passed to `rustywind`?

It should be noted that the Neovim [conform](https://github.com/stevearc/conform.nvim) formatting plugin supports `rustywind`. Conform can be setup to run common command line formatters from within Neovim, or run it automatically when saving.

Seems doable, I think.

**Comment by cirdes** [comment]
@bluz71, it’s not exactly what you’re looking for, but the result is almost the same. I have been using [tailwind-merge](https://github.com/dcastil/tailwind-merge) inside [rbui.dev](https://rbui.dev/) to merge Tailwind classes correctly. It works the same way as [tailwind-merge](https://www.npmjs.com/package/tailwind-merge), which I was using with React.
